charts/operator/serviceaccount.go

```diff
@@ -0,0 +1,59 @@
+// Licensed to the Apache Software Foundation (ASF) under one or more
+// contributor license agreements.  See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// +gotohelm:filename=_serviceaccount.go.tpl
+package operator
+
+import (
+	"github.com/redpanda-data/helm-charts/pkg/gotohelm/helmette"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Create the name of the service account to use
+func ServiceAccountName(dot *helmette.Dot) string {
+	values := helmette.Unwrap[Values](dot.Values)
+
+	if values.ServiceAccount.Create {
+		if values.ServiceAccount.Name != nil && *values.ServiceAccount.Name != "" {
```

Comment:
nit: ptr.Deref :P

Edit: Feel free to ignore this. I want to write a linter and this will be a good test case.

Reply:
Done